source/extensions/filters/http/grpc_http1_reverse_bridge/filter.cc

```diff
@@ -91,7 +97,7 @@ Http::FilterDataStatus Filter::decodeData(Buffer::Instance& buffer, bool) {
 }
 
 Http::FilterHeadersStatus Filter::encodeHeaders(Http::HeaderMap& headers, bool) {
-  if (enabled_) {
+  if (enabled_ && headers.ContentType() && headers.Status()) {
```

Comment:
I think it might make more sense to treat a missing content-type header as an error. I think it would be as easy as adding `content-type != nullptr` to the check below

As for `Status` I think that's guaranteed to be part of the encoded headers, I think the codecs will ensure this as it won't be a valid HTTP response without one (taking it either from the :status metaheader in HTTP/2 or from the HTTP/1 status line). Have you been able to see a case where the Status header is missing?

Reply:
I  can definitely change this logic to return `Http::FilterHeadersStatus::ContinueAndEndStream` if content type is nullptr. Doing the same for status makes sense too. A malicious endpoint does not need to adhere to the protocol, it could easily not include the grpc status field. I think it's safer to check than to crash, no?

Reply:
The `Status()` function actually reads the `:status` meta header, https://github.com/envoyproxy/envoy/blob/075edf802b9ef94bdf26fddc9e775f8bb89df92d/source/common/http/headers.h#L94 not the `grpc-status` header, which is `GrpcStatus`: https://github.com/envoyproxy/envoy/blob/075edf802b9ef94bdf26fddc9e775f8bb89df92d/source/common/http/headers.h#L74

I don't think the codec allows for the returned response headers to not contain the `:status` header, so I think we can safely assume it's there - unless you've observed crashes due to this not being present?

Reply:
PTAL. I think the latest commit resolves these issues!